torchvision/ops/poolers.py

```diff
@@ -165,32 +295,6 @@ def __init__(
         self.canonical_scale = canonical_scale
         self.canonical_level = canonical_level
 
-    def setup_scales(
```

Comment:
This in theory is a BC breaking change because the method was public. I don't think anyone should be calling it directly but it's worth checking on Github and FBcode to see if there are any direct calls. A safer approach could also be to expose it publicly, call the private equivalent and throw a warning.

cc @NicolasHug @fmassa thoughts?

Reply:
>  A safer approach could also be to expose it publicly, call the private equivalent and throw a warning.

I would go with that, it's safer and it's also the simplest option (we don't need to check fbcode or github). We can mark those with a ` # TODO: deprecate eventually` comment.

Similar thing could be done for https://github.com/pytorch/vision/pull/4951 as we just discussed.

Reply:
Agreed. As Nicolas pointed out in an offline, the methods `convert_to_roi_format` and `infer_scale` which were modified at #4951 need the same treatment. Aka keep a public method for the class with a call to the private one, a warning and a `TODO` comment.